packages/jellyfish-testing/src/testing.ts

```diff
@@ -6,24 +6,34 @@ import { TestingRawTx } from './rawtx'
 import { TestingICX } from './icxorderbook'
 import { TestingMisc } from './misc'
 import { TestingGroupAnchor } from './anchor'
-import { ContainerGroup, MasterNodeRegTestContainer, StartOptions } from '@defichain/testcontainers'
+import { ContainerGroup, DeFiDContainer, MasterNodeRegTestContainer, StartOptions } from '@defichain/testcontainers'
 import { RegTestFoundationKeys } from '@defichain/jellyfish-network'
 import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
 
 export class Testing {
-  public readonly fixture = new TestingFixture(this)
-  public readonly token = new TestingToken(this.container, this.rpc)
-  public readonly poolpair = new TestingPoolPair(this.container, this.rpc)
-  public readonly rawtx = new TestingRawTx(this.container, this.rpc)
-  public readonly icxorderbook = new TestingICX(this)
-  public readonly misc = new TestingMisc(this.container, this.rpc)
+  public readonly container: MasterNodeRegTestContainer
+  public readonly fixture!: TestingFixture
+  public readonly token!: TestingToken
+  public readonly poolpair!: TestingPoolPair
+  public readonly rawtx!: TestingRawTx
+  public readonly icxorderbook!: TestingICX
+  public readonly misc!: TestingMisc
 
   private readonly addresses: Record<string, string> = {}
 
   private constructor (
-    public readonly container: MasterNodeRegTestContainer,
+    public readonly con: DeFiDContainer,
     public readonly rpc: TestingJsonRpcClient
   ) {
+    this.container = this.con as MasterNodeRegTestContainer
```

Comment:
Suggestion above should avoid this casting, the arg must be masternode

Reply:
Been thinking about this. The main problem is that there are a lot of tests(old) that depend on `testing.container`  to be a MNContainer. rather than depending on the `testing` .
But we need to add the support for non MN containers one way or another.
Looks like there isn't an elegant way to handle this without a large refactor.
Keeping this on the dock until we do a proper design. and moving other important PRs forward.